.github/workflows/main.yml

```diff
@@ -3,3 +3,19 @@ name: CI
 on:
   push:
   pull_request:
+
+jobs:
+
+  tests:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - linux: py39-test
+        - macos: py310-test
+        - windows: py311-test
+
+  publish:
+    needs: tests
```

Comment:
Yeah, combining the tagging and publishing should work well, and allow the default GitHub token to be used. Each job checks out its own copy, so as long as the tagging happens before publishing it should work fine as the tag won't change the commit SHA. I tested this on a different repo. I added the `if:` to the create tag step instead of the job so the publishing (without release) is not skipped on a PR.
```suggestion
        - windows: py311-test

  # Automatically tag commits to main with the latest calendar version. Note that
  # this does not currently support two versions in the same day, but if this is
  # needed exceptionally, it is possible to manually push a tag.
  tag:
    needs: tests
    name: Tag latest commit to main using calendar version
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
      - name: Get tag name from date
        id: tag_name
        run: echo "date=$(date +'%Y.%m.%d')" > $GITHUB_OUTPUT
      - name: Check tag
        run: echo ${{ steps.tag_name.outputs.date }}
      - name: Create tag
        if: github.event_name == 'push' && github.ref_name == 'main'
        uses: mathieudutour/github-tag-action@v6.1
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          custom_tag: ${{ steps.tag_name.outputs.date }}

  publish:
    needs: tag
```